xray/backends/common.py

```diff
@@ -42,6 +46,28 @@ def is_trivial_index(var):
     return np.all(var.values == arange)
 
 
+def robust_getitem(array, key, catch=Exception, max_retries=6,
```

Comment:
It didn't seem worth adding a new dependency to xray. And with the time it would take to understand that implementation, it was just as easy to write something from scratch for the particular case I wanted.